### PR TITLE
[feature](backup) backup and restore privileges catalogs workloadgroup

### LINKED ISSUE #43952: [Feature] Support backup and restore privileges
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Support backup and restore privileges

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justfortaste**
### What problem does this PR solve?

wiki: https://doc.weixin.qq.com/doc/w3_AGkArQbjACc5rBcdFOxQcaBSzESxW?scode=AJEAIQdfAAoRB9aawYAGkArQbjACc
Issue Number: close #43952

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 
    https://github.com/apache/doris-website/pull/1442

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by justfortaste** [comment]
run buildall

**Review by lide-reed — DISMISSED** [review]
LGTM

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/FeMetaVersion.java [hunk 1]
```diff
@@ -103,8 +103,11 @@ public final class FeMetaVersion {
 
     public static final int VERSION_140 = 140;
 
+    // For backup global info
+    public static final int VERSION_141 = 141;
+
     // note: when increment meta version, should assign the latest version to VERSION_CURRENT
-    public static final int VERSION_CURRENT = VERSION_140;
+    public static final int VERSION_CURRENT = VERSION_141;
```
Comment (yiguolei) [inline fe/fe-common/src/main/java/org/apache/doris/common/FeMetaVersion.java:110]: changed fe meta version， it is very dangerous

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/FeMetaVersion.java (hunk 1 above)
Comment (justfortaste) [reply fe/fe-common/src/main/java/org/apache/doris/common/FeMetaVersion.java:110]: BackupMeta has changed,  upgrade the fe meta verison to  compatible with old versions.

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

**Review by lide-reed — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 2]
```diff
@@ -1263,6 +1298,154 @@ private void checkAndRestoreResources() {
...
+            if (Env.getCurrentEnv().getAuth().doesUserExist(user.getUserIdentity())) {
+                localUsers.add(user.getUserIdentity());
+                continue;
+            }
+            try {
+                Env.getCurrentEnv().getAuth().createUserInternal(user.getUserIdentity(), null,
+                        user.getPassword().getPassword(), true, PasswordOptions.UNSET_OPTION,
+                        user.getComment(), UUID.randomUUID().toString(), false);
+            } catch (DdlException e) {
+                LOG.error("restore user fail should not happen", e);
```
Comment (xy720) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:1316]: set status and return here, and roll back in cancelInternal

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 3]
```diff
@@ -1263,6 +1315,205 @@ private void checkAndRestoreResources() {
...
+                status = new Status(ErrCode.COMMON_ERROR, "restore sqlBlockRule "
+                        + sqlBlockRule.getName() + " failed:" + e.getMessage());
+                return;
+            }
+            restoredSqlBlockRules.add(sqlBlockRule);
+        }
+
+        List<Role> roles = jobInfo.newBackupObjects.backupGlobalInfo.getRoleList();
+        for (Role role : roles) {
+            Role oldRule = Env.getCurrentEnv().getAuth().getRoleByName(role.getRoleName());
```
Comment (zddr) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:1437]: oldRule==>oldRole?

File: regression-test/suites/backup_restore/test_backup_restore_priv.groovy [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+    sql "drop catalog if exists mysql;"
+    sql "drop workload group if exists wg1;"
+    sql "drop workload group if exists wg2;"
+
+
+    sql "CREATE USER 'user1' IDENTIFIED BY '12345' PASSWORD_EXPIRE INTERVAL 10 DAY FAILED_LOGIN_ATTEMPTS 3 PASSWORD_LOCK_TIME 1 DAY;"
+    sql "CREATE USER 'user2' IDENTIFIED BY '12345' PASSWORD_EXPIRE INTERVAL 10 DAY FAILED_LOGIN_ATTEMPTS 3 PASSWORD_LOCK_TIME 1 DAY;"
+    sql "CREATE USER 'user3' IDENTIFIED BY '12345' PASSWORD_EXPIRE INTERVAL 10 DAY FAILED_LOGIN_ATTEMPTS 3 PASSWORD_LOCK_TIME 1 DAY;"
+
+    sql "create role role_select;"
```
Comment (zddr) [inline regression-test/suites/backup_restore/test_backup_restore_priv.groovy:73]: add `col priv` case ;
add `grant priv to user` case

**Comment by justfortaste** [comment]
run buildall

**Comment by justfortaste** [comment]
runbuild all

**Review by lide-reed — APPROVED** [review]
LGTM

**Review by lide-reed — APPROVED** [review]
LGTM

**Review by liutang123 — APPROVED** [review]

**Comment by liutang123** [comment]
run buildall

**Comment by w41ter** [comment]
@justfortaste Please update the doris_website.

**Pull request merged**